pkgs/development/haskell-modules/configuration-ghc-8.2.x.nix

```diff
@@ -43,9 +44,35 @@ self: super: {
   text = self.text_1_2_3_1;
 
   # Make sure we can still build Cabal 1.x.
-  Cabal_1_24_2_0 = overrideCabal super.Cabal_1_24_2_0 (drv: {
-    prePatch = "sed -i -e 's/process.*< 1.5,/process,/g' Cabal.cabal";
-  });
+  Cabal_1_24_2_0 = self.callPackage
```

Comment:
What is the point of that change? I don't see what this is supposed to accomplish?

Reply:
@peti `super.Cabal_1_24_2_0` is not a thing anymore (I guess it was in hackage_packages sometime ago, but not now). Without this change `nix-env -qaP haskell.packages.ghc822` fails. 

I'd simply remove this derivation, but
1. the comment above says "# Make sure we can still build Cabal 1.x." 🤷‍♂️ 
2. removing this derivation is a breaking change _if_ someone depends on it and\or cares.

So, I fixed this `super.Cabal_1_24_2_0` issue and adjusted `prePatch` section so that Cabal successfully builds.

Does this answer your question?

P.S. If no one cares about `pkgname_{version}` derivations in ghc_8.x configurations, I can just remove them.

Reply:
Just drop the override. It's been broken for a long time and doubt anyone cares about Cabal 1.x any more.